Add peering endpoint API docs (#13454)
hashicorp/consul: website/content/api-docs/peering.mdx

---
layout: api
page_title: Cluster Peering - HTTP API
description: The /peering endpoints allow for managing cluster peerings.
---

# Cluster Peering - HTTP API

> ~> This page covers features that are currently in _technical preview_. Features and
> functionality are subject to change. You should never use the technical preview release
> in secure environments or production scenarios. Features in technical preview may face
> performance and scaling issues, with limited support options available.

Reviewer: ```suggestion
~> **Cluster peering is currently in technical preview:** Functionality associated with cluster peering is subject to change. You should never use the technical preview release in secure environments or production scenarios. Features in technical preview may have performance issues, scaling issues, and limited support.
```

Reviewer: this is what we have for the normal cluster peering docs

[required ACLs](/api#authentication).

| Blocking Queries | Consistency Modes | Agent Caching | ACL Required     |
| ---------------- | ----------------- | ------------- | ---------------- |
| `NO`             | `none`            | `none`        | `none`           |

### JSON Request Body Schema

- `PeerName` `(string: <required>)` - The name to assign to the peer cluster.
  The peer name is used to reference the peer cluster in service discovery queries
  and configuration entries such as `service-intentions`. This field must be a
  valid DNS hostname label.

- `Partition` `(string: "")` - <EnterpriseAlert inline /> The admin partition that the
  peering token will be generated from. Will default to `default` if not specified.

- `Datacenter` `(string: "")` - Specifies the datacenter, which defaults to the
  agent's datacenter if not provided.

- `Token` `(string: "")` - Specifies the ACL token to use in the request instead of
  the one specified via the `token` query parameter, `X-Consul-Token` request header,
  or `CONSUL_HTTP_TOKEN` environment variable.

- `Meta` `(map<string|string>: <optional>)` - Specifies arbitrary KV metadata

>   to associate with the peering.

Author: These metadata pairs are effectively labels that may have meaning to the user but not to Consul.

I added the sentence you suggested.

### Query Parameters

- `partition` `(string: "")` <EnterpriseAlert inline /> - Specifies the partition of the peering
  to read. If not specified will default to `default`.

### Sample Request

```shell-session
$ curl --header "X-Consul-Token: b23b3cad-5ea1-4413-919e-c76884b9ad60" \
   http://127.0.0.1:8500/v1/peering/cluster-02
```

### Sample Response

```json
{
    "ID": "462c45e8-018e-f19d-85eb-1fc1bcc2ef12",
    "Name": "cluster-02",
    "State": "INITIAL",
    "PeerID": "e83a315c-027e-bcb1-7c0c-a46650904a05",
    "PeerServerName": "server.dc1.consul",
    "PeerServerAddresses": [
        "10.0.0.1:8300"
    ],
    "CreateIndex": 89,
    "ModifyIndex": 89
}
```

## Delete a Peering

This endpoint marks a peering for deletion. Once marked, Consul will
delete all the data imported from the peer in the background. Only once
all associated data has been deleted will the peering actually disappear.
Until then, further reads can be performed on the peering and a `DeletedAt`
field will now be populated with the timestamp of when the peering was
marked for deletion.

| Method   | Path               | Produces |
| -------- | ------------------ | -------- |

> | `DELETE` | `/peering/:name`   | N/A      |

Author: Yep, the colon indicates that the name here is provided by the user

   --header "X-Consul-Token: b23b3cad-5ea1-4413-919e-c76884b9ad60" \
   http://127.0.0.1:8500/v1/peering/cluster-02
```

### Sample Read Output After Deletion Prior to Removal

```json
{
    "ID": "462c45e8-018e-f19d-85eb-1fc1bcc2ef12",
    "Name": "cluster-02",
    "State": "TERMINATED",
    "PeerID": "e83a315c-027e-bcb1-7c0c-a46650904a05",
    "PeerServerName": "server.dc1.consul",
    "PeerServerAddresses": [
        "10.0.0.1:8300"
    ],
    "DeletedAt": "2022-12-14T23:00:00Z",
    "CreateIndex": 89,
    "ModifyIndex": 89
}
```

## List all Peerings

This endpoint lists all the peerings.

@include 'http_api_results_filtered_by_acls.mdx'

| Method | Path          | Produces           |
| ------ | ------------- | ------------------ |
| `GET`  | `/peerings`   | `application/json` |

The table below shows this endpoint's support for
[blocking queries](/api-docs/features/blocking),
[consistency modes](/api-docs/features/consistency),
[agent caching](/api-docs/features/caching), and
[required ACLs](/api#authentication).

| Blocking Queries | Consistency Modes | Agent Caching | ACL Required  |
| ---------------- | ----------------- | ------------- | ------------- |

> | `YES`            | `consistent`      | `none`        | `none`        |

Reviewer 2: ```suggestion
| `NO`            | `consistent`      | `none`        | `none`        |
```
